### ISSUE TITLE
cache.bolt isn't in config path by default

### ISSUE THREAD

**Issue by Nothing4You** [plexdrive/plexdrive#393]
It looks like this was supposed to be fixed in #359 by @felixbuenemann.
This doesn't seem to be the case however.
I'm currently running plexdrive 5.1.0.r32.g09c8ee8 (09c8ee8) on linux.
When starting up plexdrive with a custom config folder it creates a fresh `cache.bolt` in `~/.plexdrive`.

Parsed config:
```
user@host:~$ plexdrive -c .plexdrive-foo -v 3 mount .plexdrive
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : verbosity            : DEBUG
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : root-node-id         : root
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : drive-id             :
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : config               : .plexdrive-foo
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : cache-file           : /home/user/.plexdrive/cache.bolt
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : chunk-size           : 10M
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : chunk-load-threads   : 3
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : chunk-check-threads  : 3
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : chunk-load-ahead     : 5
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : max-chunks           : 12
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : refresh-interval     : 1m0s
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : fuse-options         :
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : UID                  : 1000
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : GID                  : 1000
[PLEXDRIVE] [2021-02-15 22:59] DEBUG  : umask                : ----------
```

I believe the reason for this might be that when `argConfigPath` is read in https://github.com/plexdrive/plexdrive/blob/09c8ee8f5466d2c65cac395c428389b413b0bd6b/main.go#L51 it still contains the default value as the flags were not parsed yet, which only happens at the end of the flag definitions in https://github.com/plexdrive/plexdrive/blob/09c8ee8f5466d2c65cac395c428389b413b0bd6b/main.go#L64

I'm not sure what would be the best approach here, it doesn't look like you could just parse flags twice as that breaks help.
Trying to parse the flags twice results in the first parse to return `--help` output and exiting.
Upstream code comments also suggest that this would not be supported, at least here:
https://github.com/ogier/pflag/blob/45c278ab3607870051a2ea9040bb85fcb8557481/flag.go#L582-L583

Ignoring the `--help` issue, it would however fix the intended configuration:
```
./plexdrive -c ~/.plexdrive-test -v 3 mount ~/mnt
[PLEXDRIVE] [2021-02-16 00:10] DEBUG  : verbosity            : DEBUG
[PLEXDRIVE] [2021-02-16 00:10] DEBUG  : root-node-id         : root
[PLEXDRIVE] [2021-02-16 00:10] DEBUG  : drive-id             :
[PLEXDRIVE] [2021-02-16 00:10] DEBUG  : config               : /home/user/.plexdrive-test
[PLEXDRIVE] [2021-02-16 00:10] DEBUG  : cache-file           : /home/user/.plexdrive-test/cache.bolt
```

Looking at the code (though I'm not that familiar with golang) it doesn't seem to easily support parsing flags twice.

One option may be to check after flags parsing if the config value is changed, but this flags lib doesn't appear to support checking if a flag was set or is just default, see also https://github.com/ogier/pflag/blob/73e519546fc0bce0c395610afcf6aa4e5aec88eb/flag.go#L142-L148

Alternatively, https://github.com/spf13/pflag is an actively maintained fork of ogier's pflag, and he already in 2018 [suggested to switch to that fork](https://github.com/ogier/pflag/issues/33#issuecomment-395542126).

In this fork it looks like it can be determined whether the flag was passed by the user by checking for `Flag.Changed`:
https://github.com/spf13/pflag/blob/85dd5c8bc61cfa382fecd072378089d4e856579d/flag.go#L177
There's also a discussion about this in https://github.com/spf13/pflag/issues/293

The only downside I can see with this approach would be that supplying `-c` to `--help` would no longer update the `cache.bolt` default location, but I think that is negligible, although the help text should be adjusted a bit to reflect that this defaults to sit inside the config directory.

**Comment by felixbuenemann** [comment]
Sounds reasonable. Can you come up with a PR?

**Comment by Nothing4You** [comment]
hi, yes, until you call `flag.Parse()` `argConfigPath` will keep its default value.

**Comment by felixbuenemann** [comment]
OK, fixed in #407.

**Comment by felixbuenemann** [comment]
Released 5.2.1 with the bugfix.
